ccm_web/server/src/api/api.service.ts

```diff
@@ -37,66 +41,92 @@ export class APIService {
 
   async getCourseSections (user: User, courseId: number): Promise<CanvasCourseSection[] | APIErrorData> {
     const requestor = await this.canvasService.createRequestorForUser(user, '/api/v1/')
+    const courseHandler = new CourseApiHandler(requestor, courseId)
+    return await courseHandler.getSections()
+  }
+
+  async getCourseSectionsInTermAsInstructor (
+    user: User, termId: number
+  ): Promise<CourseWithSections[] | APIErrorData> {
+    const requestor = await this.canvasService.createRequestorForUser(user, '/api/v1/')
+    let courses: CanvasCourse[] = []
     try {
-      const endpoint = `courses/${courseId}/sections`
-      const queryParams = { include: ['total_students'] } // use list for "include" values
+      const endpoint = 'courses'
+      const queryParams = { enrollment_type: 'teacher' }
       logger.debug(`Sending request to Canvas (get all pages) - Endpoint: ${endpoint}; Method: GET`)
-      const sectionsFull = await requestor.list<CanvasCourseSection>(endpoint, queryParams).toArray()
+      courses = await requestor.list<CanvasCourse>(endpoint, queryParams).toArray()
       logger.debug('Received response (status code unknown)')
-      return sectionsFull.map(s => ({
-        id: s.id,
-        name: s.name,
-        total_students: s.total_students
-      }))
     } catch (error) {
       const errResponse = handleAPIError(error)
       return { statusCode: errResponse.canvasStatusCode, errors: [errResponse] }
     }
+
+    const coursesInTerm = courses.filter(c => c.enrollment_term_id === termId)
+    const apiPromises = coursesInTerm.map(async c => {
+      const courseHandler = new CourseApiHandler(requestor, c.id)
+      return await courseHandler.getSectionsWithCourse(CourseApiHandler.slimCourse(c))
+    })
+    const coursesWithSectionsResult = await Promise.all(apiPromises)
+    return makeResponse<CourseWithSections>(coursesWithSectionsResult)
   }
 
-  async getCourseName (user: User, courseId: number): Promise<CanvasCourseBase | APIErrorData> {
+  async getCourseSectionsInTermAsAdmin (
```

Comment:
@ssciolla I think the sections from the API `/api/v1/accounts/:account_id/courses` from the API will pull all state like `created, claimed, available, completed` . We should document this here just for  clarity in code or swagger API( leave it if adds more work) 

I feel in the testing Melinda can decide which list to populate. Courses in canvas are create 6 month before the term. Who know when an instructor will decide to merge section. I don't think any action is needed here now but sharing some insights

Reply:
Okay, would you like me to add a comment about the states returned by default? I can see maybe removing `completed` from that list, otherwise, I think the others are probably okay. Maybe we should ask Melinda?

Reply:
@melindakraft What are your thoughts on this?

Reply:
Maybe `completed` isn't really a problem since we're limiting by term. While I don't think people would really want to cross-list for past terms (where you'd have `completed` courses), I could see someone doing that for testing. In general, you wouldn't see too many `completed` courses in the current or future terms, so it wouldn't be confusing overly confusing. Also, I could imagine an instructor "complete"ing a course accidentally (instead of keeping it un-published) and still want to crosslist sections from it.

Reply:
Yes. I agree on the accidentally marking Conclude the course. I don't know conclude or complete same thing. But We can show the completed course. No harm. IMO

Reply:
Okay, @pushyamig, see comment for requested documentation: https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/199/commits/774571d19053c4057a3bfb1ba9ede1e4c7bcc97f

Reply:
Thanks for the comments

Reply:
@ssciolla @pushyamig For what it is worth, when a course is concluded, all enrollments are archived, including the instructor's access (technically, they are still there but in an inactive state). According to Canvas, I think we can determine that completed = concluded: https://community.canvaslms.com/t5/Instructor-Guide/What-are-the-different-states-of-a-Canvas-course/ta-p/1184

I don't think we should show concluded/completed courses for the purpose of section merging. If a course is accidentally concluded, there will be no section or enrollments. it is easy for an admin in Canvas to unconclude a course, though that won't put the enrollments back. That step has to be done separately. I would not expect CCM to fill that gap since it is an edge case. 

Regarding crosslisting across terms, it comes up only occasionally and can be handled outside of CCM. It is an edge case, too.

Reply:
Okay, let me add the state parameter then for everything but `deleted` and `completed`, and test before merging. Thanks for the input!

Reply:
Okay, see https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/199/commits/3c03dc6314c08fd7ef15d3b2bb7b6070c809a63f

Reply:
The new code make sense, did not test though